database-mysql/dummyDataNew.js

```diff
@@ -339,57 +342,254 @@ function Data() {
+//         if (err) {
+//           console.log(err);
+//         } else {
+//           // console.log('added: ', count)
+//           // const used = process.memoryUsage().heapUsed / 1024 / 1024;
+//           // console.log(`The script uses approximately ${Math.round(used * 100) / 100} MB`);
+//           // if (count === 100 || count === 1000 || count % 10000 === 0 || count === 100000 || count === 1000000) {
+//           //   const used = process.memoryUsage().heapUsed / 1024 / 1024;
+//           //   console.log(`The script uses approximately ${Math.round(used * 100) / 100} MB`);
+//           //   console.log('count: ', count);
+//           // }
+//           if (count === 10000) {
+//             console.log ('reached: ', count);
+//             var today = new Date();
+
+//             var minutes = ((today.getHours() * 60) + today.getMinutes()) - startMinutes;
+//             var seconds = today.getSeconds() - startSeconds;
+
+//             console.log('time elapsed:', minutes, 'm : ', seconds, 's');
+
+//           }
+//           resolve();
+//         }
+//       });
+
+//     //}, 0)
+//   });
+
+//   promise.then(function() {
+//     wrapper();
+//   });
+// }
+
+// wrapper();
+
+
+//================================
+var simulInsert = (numOfInserts, numOfRows, callback) => {
+  for (let i = 1; i < numOfInserts + 1; i++) {
+    // const used = process.memoryUsage().heapUsed / 1024 / 1024;
+    // console.log(`The script uses approximately ${Math.round(used * 100) / 100} MB`);
+    insertNumOfRows(numOfRows, () => {
+      // console.log('insert occured');
+      // console.log('current insert: ', i)
+      // console.log('max insert: ', numOfInserts)
+      if (i === numOfInserts) {
+        // console.log('about to resolve')
+        // console.log('this is the last insert : ', i,'/', numOfInserts)
+        // console.log('i: ', i);
+        callback();
+      } 
+    });
+  }
+}
+//=============================
+
+
+
 var today = new Date();
 var startMinutes = ((today.getHours() * 60) + today.getMinutes());
 var startSeconds = today.getSeconds();
+var startMilliseconds = today.getMilliseconds();
 
 
 var count = 0;
 var totalSize = 100000;
 var batchSize = 10000;
+var check = true;
 var wrapper = () => {
   if(count >= totalSize) {
     console.log('count complete: ', count);
+    var today = new Date();
+    var minutes = ((today.getHours() * 60) + today.getMinutes()) - startMinutes;
+    var seconds = today.getSeconds() - startSeconds;
+    var milliseconds = today.getMilliseconds() - startMilliseconds;
+    if (seconds < 0) {
+      minutes -= 1;
+      seconds = 60 + seconds;
+    }
+    if (milliseconds < 0) {
+      seconds -= 1;
+      milliseconds = 999 + milliseconds;
+    }
+    
+    console.log('time elapsed:', minutes, 'm : ', seconds, 's : ', milliseconds, 'ms');
```

Comment:
can use console.time('anything') and console.timeEnd('anything') to track run times